functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/jobs/ScheduledJobSpec.groovy

```diff
@@ -29,22 +29,27 @@ class ScheduledJobSpec extends SeleniumBase{
         setupProject(projectName)
         ActivityPage activityPage = page ActivityPage
         JobListPage jobListPage = page(JobListPage)
-        jobListPage.loadJobListForProject(projectName)
-        def jobXml = JobUtils.generateScheduledJobsXml("scheduledJob", "<time hour='*' seconds='*/5' minute='*' />")
+        def jobXml = JobUtils.generateScheduledJobsXml("scheduledJob", "<time hour='*' seconds='*/2' minute='*' />")
         JobUtils.createJob(projectName, jobXml, client)
+
         when:
+        jobListPage.loadJobListForProject(projectName)
         jobListPage.go()
         jobListPage.validatePage()
-        waitFor(ExecutionUtils.Retrievers.executionsForProject(client, projectName), {it.size() >= 1}, WaitingTime.EXCESSIVE )
+
         activityPage.loadActivityPageForProject(projectName)
         activityPage.go()
 
+        def refreshActivityPageAndGetExecutionCount = {
```

Comment:
This may be your intention but it looks like this is also changing what this test is testing?

The `ActivityPage` has that auto-refresh functionality built into it where it is loading new executions in the background. With this change the test itself will be performing the refresh rather than letting the application do so...

Reply:
That's a good point. It helps to understand what exactly are we testing here.
Per the test description:
> Schedules a job and checks if with time passing, executions reflects in activity section.

I've never treated this as a test of the auto-refresh functionality. 
Only as a test of "executions will show up in UI eventually".
IMO, this alligns with the pre-refactoring behaviour -- waiting for the backend to have executions (line 38) prior to loading the UI. I don't think the refresh was tested prior to the refactoring.